File: frontend/src/modules/graph/application/use-cases/discardFile.ts
import { WorkingChangesWriter } from "@modules/graph/application/contracts/WorkingChangesWriter";

export async function discardFile(
  workingChangesWriter: WorkingChangesWriter,
  repositoryPath: string,
  filePath: string
): Promise<void> {
  return workingChangesWriter.discardFile(repositoryPath, filePath);
}

File: backend/src/app/working_changes/discard_file.rs
use std::path::Path;

use crate::app::working_changes::contracts::WorkingChangesWriter;
use crate::domain::errors::AppError;

pub fn discard_file(
    working_changes_writer: &dyn WorkingChangesWriter,
    repository_path: &Path,
    file_path: &str,
) -> Result<(), AppError> {
    working_changes_writer.discard_file(repository_path, file_path)
}
